node/src/components/fetcher.rs

```diff
@@ -522,6 +522,41 @@ impl ItemFetcher<BlockHeaderWithMetadata> for Fetcher<BlockHeaderWithMetadata> {
     }
 }
 
+impl ItemFetcher<BlockSignatures> for Fetcher<BlockSignatures> {
```

Comment:
I think we should have a separate struct for this particular fetcher - mostly b/c we want to change the notion of _sufficient_ here and return if we have more than `ftt` and not `(1 + ftt)/2`. i.e. not every context where we want to fetch signatures expects the same weight of signatures.

This can be done in a separate PR.

Reply:
I will consider this approach when working on https://github.com/casper-network/casper-node/issues/3082

Reply:
> The check allowing for `ftt` signatures will be in another place - we will be validating that we have at least this many after we've tried all the peers and still haven't collected `(1+ftt)/2`

Ah, you're right. Not an issue then.